### ISSUE TITLE
Type inferrence fails in release build when using str()

### ISSUE THREAD

**Issue by EIREXE** [godotengine/godot#61092]
### Godot version

3.x ddc14b6bbce41ca539e6314403b5afea7d192b19

### System information

Arch Linux

### Issue description

This error is not the same as #59723

example code:

```
var item_id := 0

func _ready():
	var id := "ugc_" + str(item_id)
```

### Steps to reproduce

Execute the code above

### Minimal reproduction project

[type_inference_mrp.zip](https://github.com/godotengine/godot/files/8701236/type_inference_mrp.zip)

**Comment by EIREXE** [comment]
> Is this a regression or also reproducible on stable releases?

Just checked on stable 3.4.4, doesn't seem to happen.

**Comment by akien-mga** [comment]
Fixed by #61117.
